packages/html-reporter/src/reportView.tsx

```diff
@@ -104,19 +126,43 @@ const TestCaseViewLoader: React.FC<{
       }
     })();
   }, [test, report, testId, testIdToFileIdMap]);
-  return <TestCaseView projectNames={report.json().projectNames} test={test} anchor={anchor} run={run}></TestCaseView>;
+
+  return (
+    <TestCaseView
+      projectNames={report.json().projectNames}
+      test={test}
+      anchor={anchor}
+      run={run}
+      prevTestId={prevTestId}
+      nextTestId={nextTestId}
+    />
+  );
 };
 
-function computeStats(files: TestFileSummary[], filter: Filter): FilteredStats {
+
+function computeStats(filteredFiles: TestFileSummary[]): FilteredStats {
   const stats: FilteredStats = {
     total: 0,
     duration: 0,
   };
-  for (const file of files) {
-    const tests = file.tests.filter(t => filter.matches(t));
-    stats.total += tests.length;
-    for (const test of tests)
+  for (const file of filteredFiles) {
+    stats.total += file.tests.length;
+    for (const test of file.tests)
       stats.duration += test.duration;
   }
   return stats;
-}
\ No newline at end of file
+}
+
+function getAdjacentTestIds(testIdToFileIdMap: Map<string, string>, currentTestId: string) {
+  const testIds = [...testIdToFileIdMap.keys()];
+  const currentIndex = testIds.indexOf(currentTestId);
+
+  const lastIndex = testIds.length - 1;
+  const nextIndex = currentIndex === lastIndex ? 0 : currentIndex + 1;
```

Comment:
I don't think you should look them for better UX, make it return undefined and disable the button instead. That way I know I went through all.